src/pz390.java

```diff
@@ -8129,12 +8133,40 @@ private void ins_B9XX(){
 				reg.putLong(rf1, rlv3);
 				psw_cc = get_long_log_sub_cc();
 				break;
-	     case 0xEC: // "B9EC=MGRK,54,340", // B9EC rrfa MGRK R1,R2,R3 RPI 2202
+	     case 0xEC: // "B9EC=MGRK,54,340", // B9EC rrfa MGRK R1,R2,R3 RPI 2202 #300
+		     psw_check = false;
 	    	 ins_setup_rrfa();
+			 if ((mf1 & 1) != 0){ // issue 300 add MGRK, MSGRKC, MSRKC
+				set_psw_check(psw_pic_spec);
+			}
+			big_int1 = new BigInteger(get_signed_bytes(reg_byte, rf2, 8)); 
+																			
+			big_int2 = new BigInteger(get_signed_bytes(reg_byte, rf3, 8));
+			big_int1 = big_int1
+			         .multiply(big_int2);
+			fp_bi_to_wreg(reg_byte,rf1,big_int1, 16); 
 	    	 break;
-	    case 0xED: // "B9ED=MSGRKC,54,340", // B9ED rrfa MSGRKC R1,R2,R3 RPI 2202
+	    case 0xED: // "B9ED=MSGRKC,54,340", // B9ED rrfa MSGRKC R1,R2,R3 RPI 2202 #300
+			 psw_check =false;
 	    	 ins_setup_rrfa();
-	    	 break;
+			 big_int1 = new BigInteger(get_signed_bytes(reg_byte, rf2, 8)); 
+																			
+			big_int2 = new BigInteger(get_signed_bytes(reg_byte, rf3, 8));
+			big_int1 = big_int1
+			         .multiply(big_int2);
+
+       fp_bi_to_wreg(reg_byte,rf1,big_int1, 8);
+			 if (big_int1.compareTo(bi_max_pos_long) == 1
+			     || big_int1.compareTo(bi_min_neg_long) == -1) {                  
+			     psw_cc = psw_cc3;                          
+			 } else if (big_int1.compareTo(BigInteger.ZERO) == 1){
+                                  psw_cc = psw_cc2;
+                              } else if (big_int1.compareTo(BigInteger.ZERO) == -1){
+                                  psw_cc = psw_cc1;
+                              } else {
+                                  psw_cc = psw_cc0;
+			} 
```

Comment:
Suggestion only. You could replace this else if block with
```
			} else {
				switch (big_int2.compareTo(BigInteger.ZERO))
				{
				case 1:
					psw_cc = psw_cc2;
					break;
				case -1:
					psw_cc = psw_cc1;
					break;
				default:
					psw_cc = psw_cc0;
					break;
				}			
```

Reply:
I had a typo in the switch statement -- big_int2 instead of the correct big_int1. Fixed the comment, but added this one in case you already looked at it. Sorry about that.